### DISCUSSION TITLE
Can I use this in commerical projects?

### DISCUSSION THREAD

**Discussion by hsuraweera** [tinymce/tinymce#8539] [General]
Hi, I need to clarify that is it possible to use the self-hosted free version in a commercial project please?

**Comment by raujohn** [comment]
Sure is! TinyMCE is available under MIT license. If you want to do more, you can also check out our premium capabilities, which also fall under a commercial license: https://www.tiny.cloud/tinymce/features/

**Comment by mwaddell** [comment]
Not anymore :-(  The self-hosted free version of v6 becomes EOL in October and v7+ is no longer free.

**Reply by lostkeys** [reply]
Hi @mwaddell, TinyMCE 7 is free to use under the GPL license. Quoting [Wikipedia](https://en.wikipedia.org/wiki/GNU_General_Public_License)
>The GNU General Public License (GNU GPL or simply GPL) is a series of widely used [free software licenses](https://en.wikipedia.org/wiki/Free_software_license), or [copyleft](https://en.wikipedia.org/wiki/Copyleft), that guarantee [end users](https://en.wikipedia.org/wiki/End_user) the four freedoms to run, study, share, and modify the software.

**Reply by mwaddell** [reply]
Not in commercial products

**Reply by mwaddell** [reply]
@lostkeys Quoting from the GPL:

```
You must cause any work that you distribute or publish, that in whole 
or in part contains or is derived from the Program or any part thereof,
to be licensed as a whole at no charge to all third parties under the 
terms of this License.
```

So any work that contains TinyMCE must now be GPL as well. This is incompatible with most commercial software as well as any open-source software that uses more commercial-friendly licenses (MIT, BSD, etc).

This wasn't a problem when TinyMCE was under LGPL (or MIT) but it is now under the GPL.

**Reply by mwaddell** [reply]
@lostkeys As a small IT consulting company, we've developed several web applications for our clients over the years that use TinyMCE (some that are publicly accessible on the web and some that are deployed as internal business applications). 

Moving forward, we will not be allowed to do this without either (a) only using the cloud-hosted version and having each of our customers sign up for their own TinyMCE license so they can be individually billed for any "loads" beyond 1000/month, or (b) only using the self-hosted version and telling our customers that the source code for the product they paid us to develop for them has to be published for free on the internet.

Neither of those options is likely to be acceptable, which means we will need to look for an alternative.

This also means that any existing applications we have developed cannot be upgraded beyond v6 (even if a security issue is found in v6) without changing to one of the options listed above.

**Reply by mwaddell** [reply]
@lostkeys Also understand that I am not at all opposed to paying for high-quality software as long as the licensing and pricing models are consistent with our business model and our clients' expectations.  

We currently pay for other software libraries and tools with licensing and pricing models friendly to small businesses (e.g. a yearly per-developer or per-build-server fee that covers unlimited self-hosted deployments for our clients).

TinyMCE's new pricing model is an attempt to force anyone currently hosting the TinyMCE javascript library along with the rest of an application to instead pay separately for "TinyMCE CDN hosting" at an inflated rate ($130 for 20,000 operations with TinyMCE CDN Hosting vs $0.015 for 20,000 operations at Google Cloud CDN).

We use Google Maps for some of our solutions. Our clients understand that there is a free usage level and they will pay extra if they exceed it.  They also understand the risk that if Google Maps goes offline, deprecates an API version, or has a major security issue, they will be impacted. Our clients understand that Google Maps is not something that is (or could be) contained entirely within our application and that it's an entirely separate service that exists outside of our application.

Unlike Google Maps, TinyMCE is not an API to an externally hosted service, it is a library used to build applications.  The end user of an application that uses TinyMCE has no reason to expect that their application will break if they go offline (for example if TinyMCE is used in an Electron-based desktop application) and no reason to believe it is not hosted alongside the rest of their application code.

**Reply by carlosmintfan** [reply]
However, if you're using Wordpress, then GPL-licensed TinyMCE shouldn't be that a problem as WP itself is also under GPL and therefore also puts restrictions on your use :)

**Reply by TheSpyder** [reply]
> Moving forward, we will not be allowed to do this without either (a) only using the cloud-hosted version and having each of our customers sign up for their own TinyMCE license so they can be individually billed for any "loads" beyond 1000/month, or (b) only using the self-hosted version and telling our customers that the source code for the product they paid us to develop for them has to be published for free on the internet.

@mwaddell we do offer a self-hosted commercial license, although I'm not in sales so I can't help you with any potential cost figures for that. I believe we have existing VAR relationships with businesses such as yours, but you'd need to contact the sales team to discuss whether there are options more suited to your deployments.

**Reply by carlosmintfan** [reply]
@mwaddell Although I'm hoping that my HugeMCE project will be a good alternative in the future, it's also important to notice that if I understand correctly, you'd not have to release your backend code (NodeJS/PHP/Python etc.) in order to be able to use GPL-licensed TinyMCE, only the frontend code, as TinyMCE is not AGPL-licensed.

**Reply by mwaddell** [reply]
@carlosmintfan According to my legal counsel, if the frontend and backend code are designed to work together as two parts of a whole application (and neither is fully functional without the other), then yes, you would need to release both under the GPL if any portion of either contains GPLed code.

AGPL is a license for standalone APIs which requires that other applications which integrate with them must also be GPLed. The AGPLed code can exist and fully function without the GPLed code while the GPLed code relies on the AGPLed code.

**Reply by carlosmintfan** [reply]
Oh no. That's stupid. Although I'm not sure that programs interacting with an AGPLed API would really need to be licensed under (A)GPL. ChatGPT also says not. But, I'm not a lawyer...
My perception of the AGPL was that if you run AGPLed code on your server, you have to give it to the users of your server although your app only gives them the output and no binaries of the program itself.

**Reply by mwaddell** [reply]
I may be misinformed about the AGPL then (or maybe lawyers disagree on how to interpret it), but I'm fairly confident that GPL of any "part" requires GPL for the "whole" unless the parts are fully distinct and can operate on their own.

**Reply by carlosmintfan** [reply]
@mwaddell Yeah, that may be true.

**Reply by mwaddell** [reply]
@TheSpyder I talked with your sales team and the lowest-cost option for self-hosted commercial products (for an application with no more than 100 total users) is $6,050/year.  

So, to answer @hsuraweera's original question; no, the self-hosted free version cannot be used in commercial projects any longer. 

For many existing products that use TinyMCE 6.x or earlier, the costs associated with migrating to an alternative library that uses a commercial-friendly open-source license (LGPL, MIT, Apache, etc.) may be less expensive than purchasing your commercial license.

**Reply by hsuraweera** [reply]
@mwaddell, can I use the self-hosted free version for internal POC projects within an organization that doesn’t generate revenue from using it?

**Reply by carlosmintfan** [reply]
@hsuraweera It's actually not forbidden to generate revenue with the free self-hosted version, it's just that you have to share the source code with everybody which is usually not good for commercial businesses. But for internal projects everything is okay, the GPL allows you to do (almost) whatever you want with the software as long as you don't publish it, and if you publish it, you'll just also have to publish the source code.
